lib/internal/fs/sync.js

```diff
@@ -88,6 +88,11 @@ function close(fd) {
   return binding.closeSync(fd);
 }
 
+function fsync(fd) {
+  fd = getValidatedFd(fd);
```

Comment:
Nit: If you moved this function to c++, it would improve the happy path (existing file)

Reply:
That's a good opinion!
I think I just need to add `CHECK_GE(fd,0)` in C++. Do you have any better ideas?

Reply:
That's a good opinion!
I think you just need to add CHECK_GE(fd,0) to C++. 
Do you have a better idea?

Reply:
We need to replicate the behavior of `getValidatedFd` in C++ and throw the correct error. 

```
const getValidatedFd = hideStackFrames((fd, propName = 'fd') => {
  if (ObjectIs(fd, -0)) {
    return 0;
  }

  validateInt32(fd, propName, 0);

  return fd;
});
```

Reply:
I added the implementation in C++.
I'm not sure if this implementation is correct.
also I updated the benchmark.
When I check it locally, the performance is strange.

Please review again.